### PR TITLE
Fix gcloud cli install and increase KubernetesPodOperator ephemeral-storage

### PR DESCRIPTION
**Pull request by raymondzouu**
# Description

Fix gcloud cli install for XPK docker images and increase KubernetesPodOperator ephemeral-storage.
* update `google-cloud-sdk` install to `google-cloud-cli` install due to b/324101663
* update `google-cloud-cli` installation to follow steps from [here](https://cloud.google.com/sdk/docs/install#deb)
* increase `ephemeral-storage` for `KubernetesPodOperator`- was seeing errors like https://screenshot.googleplex.com/8SqhcKuZFMG4wZD when trying to install packages on the docker image

# Tests
* ran `maxtext-sweep-gke-example-dag` http://shortn/_Z7TfN6Lem8

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [x] I have performed a self-review of my code.
- [x] I have necessary comments in my code, particularly in hard-to-understand areas.
- [x] I have run one-shot tests and provided workload links above if applicable. 
- [x] I have made or will make corresponding changes to the doc if needed.

### CODE REVIEW

File: xlml/utils/xpk.py [hunk 1]
```diff
@@ -58,13 +59,15 @@ def run_workload(
 
   cmds = (
       "set -x",
-      f"gcloud config set project {cluster_project}",
-      f"gcloud config set compute/zone {zone}",
       "git clone https://github.com/google/xpk.git /tmp/xpk",
       "cd /tmp/xpk",
-      "apt-get update && apt-get install -y google-cloud-sdk",
+      "curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | gpg --dearmor -o /usr/share/keyrings/cloud.google.gpg",
+      'echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" | tee -a /etc/apt/sources.list.d/google-cloud-sdk.list',
```
Comment (jonb377) [inline xlml/utils/xpk.py:65]: This is great! Now that we're installing gcloud cli in the command, can we decouple from the workload image here? Try chaning [L84](http://shortn/_Ema6NScBxX) to `image='python:3.10'`.

Comment (raymondzouu) [reply xlml/utils/xpk.py:65]: Good idea! I changed it and tested that it works.

File: xlml/utils/xpk.py [hunk 2]
```diff
@@ -78,11 +81,14 @@ def run_workload(
...
       arguments=[";".join(cmds)],
       namespace="composer-user-workloads",
-      image=docker_image,
+      image="python:3.10",
       config_file="/home/airflow/composer_kube_config",
       kubernetes_conn_id="kubernetes_default",
       startup_timeout_seconds=startup_timeout,
       owner=task_owner,
+      container_resources=k8s_models.V1ResourceRequirements(
+          limits={"ephemeral-storage": "10G"},
```
Comment (RissyRan) [inline xlml/utils/xpk.py:90]: Nice! Just wondering if you met issues without it?

Comment (RissyRan) [reply xlml/utils/xpk.py:90]: Oh.. I see the issue you mentioned. Thanks!

**Review by RissyRan — APPROVED** [review]

**Review by jonb377 — APPROVED** [review]
LGTM, thanks Raymond!

**Pull request merged**